### ISSUE TITLE
Eventual assertion

### ISSUE THREAD

**Issue by japgolly** [japgolly/test-state#10]
When a step in an action fails, there should be an option to wait X and try again until success or time exceeds Y. Invariants are exempt; they by definition should never fail.

Also it should be configurable at various levels:
* plan
* action
* check (?)

**Comment by japgolly** [comment]
Also the wait/sleep mechanism should be configurable (eg. back to the thread pool, schedule continuation). Will need to handle the impure types too :/

Probably something like 

```scala
trait Thing[F[_]] {
  def currentTime: F[Instant]
  def schedule[A](f: () => F[A], when: Instant): () => F[A]
}

// Fn to determine whether to retry or not, and for how long
// failures = NEV of times when previous attempts failed */
def retry(failures: NonEmptyVector[Instant], now: Instant): Option[Instant]
```

**Comment by japgolly** [comment]
Merged in #12
